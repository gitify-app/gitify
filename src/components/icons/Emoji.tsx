import twemoji from '@twemoji/api';
import type { FC } from 'react';

export interface IEmoji {
  emoji: string;
}

export const Emoji: FC<IEmoji> = (props: IEmoji) => {
  // Generate the HTML for the emoji using twemoji
  const emojiHtml = twemoji.parse(props.emoji, {
    folder: 'svg',
    ext: '.svg',
  });

  // biome-ignore lint/security/noDangerouslySetInnerHtml: used for cross-platform emoji compatibility
  return <span dangerouslySetInnerHTML={{ __html: emojiHtml }} />;
};
