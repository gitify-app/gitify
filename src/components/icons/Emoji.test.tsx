import { render } from '@testing-library/react';
import { Emoji, type IEmoji } from './Emoji';

describe('components/icons/Emoji.tsx', () => {
  it('should render - online CDN SVGs', () => {
    const props: IEmoji = {
      emoji: '🍺',
    };
    const tree = render(<Emoji {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render - offline SVGs', () => {
    const props: IEmoji = {
      emoji: '🛜',
    };
    const tree = render(<Emoji {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
