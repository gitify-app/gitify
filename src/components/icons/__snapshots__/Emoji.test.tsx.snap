// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/icons/Emoji.tsx should render - offline SVGs 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span>
        <img
          alt="🛜"
          class="emoji"
          draggable="false"
          src="/Users/adam/Documents/dev/GitHub/gitify-app/gitify/src/assets/svg/1f6dc.svg"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span>
      <img
        alt="🛜"
        class="emoji"
        draggable="false"
        src="/Users/adam/Documents/dev/GitHub/gitify-app/gitify/src/assets/svg/1f6dc.svg"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/icons/Emoji.tsx should render - online CDN SVGs 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span>
        <img
          alt="🍺"
          class="emoji"
          draggable="false"
          src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.1.0/assets//svg/1f37a.svg"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span>
      <img
        alt="🍺"
        class="emoji"
        draggable="false"
        src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.1.0/assets//svg/1f37a.svg"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
